R/addins.R

```diff
@@ -1,23 +1,23 @@
 # nocov start
 addin_lint <- function() {
   if (!requireNamespace("rstudioapi", quietly = TRUE)) {
-    stop("'rstudioapi' is required for add-ins.", call. = FALSE)
+    cli_abort("{.pkg rstudioapi} is required for add-ins.")
   }
   filename <- rstudioapi::getSourceEditorContext()
   if (filename$path == "") {
-    return("Current source has no path. Please save before continue")
+    return("Current source has no path. Please save before continuing.")
```

Comment:
Shouldn't this output a warning and return empty `Lints()` instead?

Reply:
You mean `return(list())`, right?

Reply:
A `list()` with class `"lints"` would be better (so that the print method is the correct one).

Reply:
Okay, I am going to create a new issue for this because currently we are returning just the empty list across many linters and so this change would be out of scope: https://github.com/r-lib/lintr/issues/2556